scylla-cql/src/frame/value_tests.rs

```diff
@@ -72,6 +72,30 @@ fn counter_serialization() {
     );
 }
 
+#[test]
+fn cql_varint_normalization() {
+    let test_cases = [
+        (vec![], vec![]),                     // 0
+        (vec![0x00], vec![0x00]),             // 0
+        (vec![0x00, 0x00], vec![0x00]),       // 0
+        (vec![0x01], vec![0x01]),             // 1
+        (vec![0x00, 0x01], vec![0x01]),       // 1
+        (vec![0x7f], vec![0x7f]),             // 127
+        (vec![0x00, 0x7f], vec![0x7f]),       // 127
+        (vec![0x80], vec![0x80]),             // -128
+        (vec![0x00, 0x80], vec![0x00, 0x80]), // 128
+        (vec![0xff], vec![0xff]),             // -1
+        (vec![0x00, 0xff], vec![0x00, 0xff]), // 255
+    ];
+
+    for test in test_cases {
+        let non_normalized = CqlVarint::from_signed_bytes_be(test.0);
+        let normalized = CqlVarint::from_signed_bytes_be(test.1);
+
+        assert_eq!(non_normalized, normalized);
```

Comment:
Would be nice to check here that these properties hold after conversion to BigInt, too.